### PR TITLE
YOLOv4 support

### PR DESCRIPTION
**Pull request by negvet**

### CODE REVIEW

**Comment by alexsu52** [comment]
@negvet,  could you provide the sota checkpoint results for the PR?

File: beta/examples/tensorflow/common/object_detection/utils/input_utils.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
+# pylint: disable=E1120
+# pylint: disable=E1124
```
Comment (daniil-lyakhov) [inline beta/examples/tensorflow/common/object_detection/utils/input_utils.py:14]: @andrey-churkin @alexsu52 is it more clear to disable pylint errors exactly in places where they appears? And using names like `# pylint: disable=no-value-for-parameter`

Comment (negvet) [reply beta/examples/tensorflow/common/object_detection/utils/input_utils.py:14]: I agree with the comment regarding the location of the `pylint: disable=...`. I put it on the top of the file because there are multiple places with the same type of `pylint: disable=...`. Although, it may be confusing.

Comment (alexsu52) [reply beta/examples/tensorflow/common/object_detection/utils/input_utils.py:14]: In exceptional cases, it is justified.

File: beta/examples/tensorflow/object_detection/configs/yolo_v4_coco.json [hunk 2]
```diff
@@ -4,9 +4,34 @@
...
+    "epochs": 30,
+    "batch_size": 66,
+    "workers": 12,
 
     "dataset": "coco/2017",
-    "dataset_type": "tfds"
+    "dataset_type": "tfds",
+
+    "optimizer": {
+        "type": "momentum",
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/object_detection/configs/yolo_v4_coco.json:15]: It is correct. `momentum` type of optimizer corresponds to the SGD optimizer with momentum.

File: beta/nncf/tensorflow/quantization/algorithm.py [hunk 3]
```diff
@@ -69,8 +69,12 @@ def __init__(self, config):
         for quantizer_group in QUANTIZER_GROUPS:
             self._parse_group_params(self.config, quantizer_group)
 
+        # temporal solution
+        self.compression_config = config.get('compression', {})
+        self.initializer_config = self.compression_config.get('initializer', {})
+
     def build_controller(self, model):
-        return QuantizationController(model)
+        return QuantizationController(model, self.initializer_config)
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/nncf/tensorflow/quantization/algorithm.py]: Fixed

File: beta/examples/tensorflow/object_detection/configs/yolo_v4_coco.json [hunk 4]
```diff
@@ -4,9 +4,34 @@
...
+                1e-6
+            ],
+            "learning_rate_steps": [
+                5000,
+                45000
+            ]
+        }
+    }
+
+
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/object_detection/configs/yolo_v4_coco.json]: Fixed

File: beta/examples/tensorflow/segmentation/preprocessing/maskrcnn_preprocessing.py [hunk 5]
```diff
@@ -306,3 +308,48 @@ def _parse_predict_data(self, data):
         }
 
         return inputs, labels
+
+    def _tfds_decoder(self, features_dict):
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/segmentation/preprocessing/maskrcnn_preprocessing.py]: Upon current update, `specicifc_model_preprocessing.py ` suppose to return two fucntions
`return self._tfds_decoder, self._pipeline_fn`
For Mask-RCNN (and all models that do not support `tfds`), then, I propose to return 
`return None, self._pipeline_fn`
Does it look good?

File: beta/examples/tensorflow/object_detection/preprocessing/yolo_v4_preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+        labels = {
+            'y_true_0': out0,
+            'y_true_1': out1,
+            'y_true_2': out2,
+        }
+
+        return image_data, labels
+
+    def _get_image_info(self, image):
+
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/object_detection/preprocessing/yolo_v4_preprocessing.py]: Fixed

File: beta/examples/tensorflow/object_detection/preprocessing/yolo_v4_preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+        if len(boxes) > max_boxes:
+            boxes = boxes[:max_boxes]
+        box_data = np.zeros((max_boxes, 5))
+        if len(boxes) > 0:
+            box_data[:len(boxes)] = boxes
+
+        return image_data, box_data
+
+    def _preprocess(self, image, groundtruth_classes, groundtruth_boxes, input_shape):
+
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/object_detection/preprocessing/yolo_v4_preprocessing.py]: Fixed

File: beta/examples/tensorflow/common/object_detection/losses.py [hunk 8]
```diff
@@ -471,3 +474,362 @@ def box_loss(self, box_outputs, box_targets, num_positives):
...
+    def softmax_focal_loss(self, y_true, y_pred, gamma=2.0, alpha=0.25):
+        """
+        Compute softmax focal loss.
+        Reference Paper:
+            "Focal Loss for Dense Object Detection"
+            https://arxiv.org/abs/1708.02002
+
+        # Arguments
+            y_true: Ground truth targets,
+                tensor of shape (?, num_boxes, num_classes).
```
[the remark this answers is not in the document]
Comment (negvet) [reply beta/examples/tensorflow/common/object_detection/losses.py]: Fixed

**Comment by alexsu52** [comment]
@negvet, please, update the readme relative to obtained results.

**Comment by negvet** [comment]
> @negvet, please, update the readme relative to obtained results.

I updated readme

**Review by alexsu52 — APPROVED** [review]

**Comment by andrey-churkin** [comment]
Jenkins please retry a build

File: beta/examples/tensorflow/segmentation/preprocessing/maskrcnn_preprocessing.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (andrey-churkin) [reply beta/examples/tensorflow/segmentation/preprocessing/maskrcnn_preprocessing.py]: Ok

**Comment by andrey-churkin** [comment]
Please rebase your branch when https://github.com/openvinotoolkit/nncf/pull/626 will be merged.
It should helps with errors in tests.

**Review by andrey-churkin — APPROVED** [review]
